lib/sqlmap/sqlmap_manager.rb

```diff
@@ -8,43 +8,56 @@ def initialize(session)
 
     def new_task
       res = @session.get('/task/new')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def delete_task(task_id)
       res = @session.get('/task/' + task_id + '/delete')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def set_option(task_id, key, value)
       post = { key => value }
       res = @session.post('/option/' + task_id + '/set', nil, post.to_json, {'ctype' => 'application/json'})
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def get_options(task_id)
       res = @session.get('/option/' + task_id + '/list')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def start_task(task_id, options = {})
       res = @session.post('/scan/' + task_id + '/start' , nil, options.to_json, {'ctype' => 'application/json'})
-      return JSON.parse(res.body)
+      parse_response(res)
+
     end
 
     def get_task_status(task_id)
       res = @session.get('/scan/' + task_id + '/status')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def get_task_log(task_id)
       res = @session.get('/scan/' + task_id + '/log')
-      return JSON.parse(res.body)
+      parse_response(res)
     end
 
     def get_task_data(task_id)
       res = @session.get('/scan/' + task_id + '/data')
-      return JSON.parse(res.body)
+      parse_response(res)
+    end
+
+    private
+    def parse_response(res)
+      if res && res.body
+        begin
+          res = JSON.parse(res.body)
+        rescue JSON::ParserError
+        end
+      end
+
+      res
```

Comment:
If it is not parseable, something has gone wrong and you can't really do anything reasonable with the data. Allowing the raised exception to stop the module is correct in that case. The module should be responsible for rescuing if it knows how to continue after such a failure.